pallets/dapp-staking-v3/src/migration.rs

```diff
@@ -38,12 +41,224 @@ pub mod versioned_migrations {
         Pallet<T>,
         <T as frame_system::Config>::DbWeight,
     >;
+
+    /// Migration V7 to V8 wrapped in a [`frame_support::migrations::VersionedMigration`], ensuring
+    /// the migration is only performed when on-chain version is 7.
+    pub type V7ToV8<T, TierThresholds> = frame_support::migrations::VersionedMigration<
+        7,
+        8,
+        v8::VersionMigrateV7ToV8<T, TierThresholds>,
+        Pallet<T>,
+        <T as frame_system::Config>::DbWeight,
+    >;
+}
+
+// TierThreshold as percentage of the total issuance
+mod v8 {
+    use super::*;
+    use crate::migration::v7::TierParameters as TierParametersV7;
+    use crate::migration::v7::TiersConfiguration as TiersConfigurationV7;
+
+    pub struct VersionMigrateV7ToV8<T, TierThresholds>(PhantomData<(T, TierThresholds)>);
+
+    impl<T: Config, TierThresholds: Get<[TierThreshold; 4]>> OnRuntimeUpgrade
+        for VersionMigrateV7ToV8<T, TierThresholds>
+    {
+        fn on_runtime_upgrade() -> Weight {
+            // 1. Update static tier parameters with new thresholds from the runtime configurable param TierThresholds
+            let result = StaticTierParams::<T>::translate::<TierParametersV7<T::NumberOfTiers>, _>(
+                |maybe_old_params| match maybe_old_params {
+                    Some(old_params) => {
+                        let tier_thresholds: Result<
+                            BoundedVec<TierThreshold, T::NumberOfTiers>,
+                            _,
+                        > = BoundedVec::try_from(TierThresholds::get().to_vec());
+
+                        match tier_thresholds {
+                            Ok(tier_thresholds) => Some(TierParameters {
+                                slot_distribution: old_params.slot_distribution,
+                                reward_portion: old_params.reward_portion,
+                                tier_thresholds,
+                            }),
+                            Err(err) => {
+                                log::error!(
+                                    "Failed to convert TierThresholds parameters: {:?}",
+                                    err
+                                );
+                                None
+                            }
+                        }
+                    }
+                    _ => None,
+                },
+            );
+
+            if result.is_err() {
+                log::error!("Failed to translate StaticTierParams from previous V7 type to current V8 type. Check TierParametersV7 decoding.");
+                return T::DbWeight::get().reads_writes(1, 0);
```

Comment:
Maybe also put the pallet into maintenance mode if this happens?

I know it won't, but since we have this code, why not be 100% safe? 🙂